pkg/common/x509util/serialnumber.go

```diff
@@ -1,23 +1,28 @@
 package x509util
 
 import (
-	"context"
+	"crypto/rand"
+	"fmt"
+	"math"
 	"math/big"
-	"sync/atomic"
 )
 
-type SerialNumber interface {
-	NextNumber(context.Context) (*big.Int, error)
-}
-
-type serialNumber struct {
-	next int64
-}
+// NewSerialNumber creates a random certificate serial number according to CA/Browser forum spec
+// Section 7.1:
+//   "Effective September 30, 2016, CAs SHALL generate non-sequential Certificate serial numbers greater than
+//   zero (0) containing at least 64 bits of output from a CSPRNG"
+func NewSerialNumber() (*big.Int, error) {
+	// Get MaxUint64 as big.Int
+	maxUInt64 := new(big.Int)
+	maxUInt64 = maxUInt64.SetUint64(math.MaxUint64)
 
-func NewSerialNumber() SerialNumber {
-	return &serialNumber{}
-}
+	// Creates random integer in range [0,MaxUint64)
+	s, err := rand.Int(rand.Reader, maxUInt64)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create random number: %v", err)
+	}
 
-func (m *serialNumber) NextNumber(ctx context.Context) (*big.Int, error) {
-	return big.NewInt(atomic.AddInt64(&m.next, 1)), nil
+	// Adds 1 to return serial number [1,MaxUint64]
+	s.Add(s, big.NewInt(1))
```

Comment:
nit: return the value from Add directly (then people don't have to look up if Add mutates `s`, which it does, but isn't directly clear)
```
return s.Add(s, one)
```